### DISCUSSION TITLE
clientManipulator local enviroment

### DISCUSSION THREAD

**Discussion by ma-gu-16** [GeniusesOfSymfony/WebSocketBundle#458] [Q&A]
Hi there, 

in my securedtopichandler im getting the current logged in user. sessions are shared with pdo

```
        // check if user authorized
        $user = $this->clientManipulator->getUser($conn);
```

on my local windows enviroment the user is anonymous instead of authorized. on my server its works well. 

any ideas ? 

websocket connect itself works well. 

thank you .

**Comment by mbabker** [comment]
Check your session settings, especially the domain for your session cookie.  The session provider in Ratchet, similar to traditional HTTP requests, relies on the presence of the session cookie to be able to load the correct session, and the authentication APIs in this bundle relies on that provider loading the right session data to load the authenticated user.

**Comment by ma-gu-16** [comment]
hi mbabker, 

my domain is sub.test.local 

framework config : 
` 
session:
    handler_id: Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler
    gc_maxlifetime: 36000
    cookie_secure: auto
    cookie_samesite: lax

`

if i check the session with chrome the domain is sub.test.local but if i set explicit the name with : cookie_domain: 'sub.test.local', even if i use placeholder. it doesnt work. i cant login anymore. 

the login works only without any configuration. but as i said the domainname should be correct. 

the PostAuthenticationGuardToken is set in my database with the correct user. 

the session_id 
`

07:40:45 INFO      [websocket] anon-1153923466618a25fd55976381124341 disconnected
[
  "connection_id" => 1799,
  "session_id" => "1153923466618a25fd55976381124341",
  "storage_id" => "1799",
  "user_identif

`

is def. wrong. dont know why rachet takes wrong session or no session. there is only 1 session available. :(

**Reply by mbabker** [reply]
How are you connecting to the websocket server?  IP address or hostname?  (Check the source code of your rendered page or the network tab on your browser’s dev tools to verify this).  If it’s by IP, you need to make sure the session cookie isn’t limited to only your domain otherwise it won’t work (that cookie won’t get passed through the websocket request).  If it’s by hostname, make sure it’s the same domain or a subdomain that the session cookie allows.

**Reply by ma-gu-16** [reply]
GOS_WEB_SOCKET_SERVER_IP=127.0.0.1
GOS_WEB_SOCKET_SERVER_PORT=1337
GOS_WEB_SOCKET_URI=ws://127.0.0.1:1337

ok, this must be the problem. the big question how can i start the server with domainname instead of ip. 

is it possible to make the symfony session available for all  , ip/hostname
